public_html/wp-content/mu-plugins/blocks/source/components/item-select/utils.js

```diff
@@ -28,6 +29,11 @@ const buildOptionGroup = ( entityType, type, label, items ) => {
 
 				parsedItem.avatar = get( item, 'avatar_urls[\'24\']', '' );
 				parsedItem.image  = get( item, '_embedded[\'wp:featuredmedia\'][0].media_details.sizes.thumbnail.source_url', '' );
+
+				if ( 'wcb_session' === item.type ) {
+					parsedItem.details = getSessionDetails( item );
+				}
```

Comment:
I'd love it if we could avoid importing functionality and adding conditionals into our reusable components that are specific to one block.

Is there possibly a reasonable way to inject the output of `getSessionDetails` into each of the items in `entities.wcb_session` (which is used as `items` here)? That way this function could just look for `item.details` without having to know about specific post types.

Reply:
Good idea, I've moved the `getSessionDetails` into the initial select call so it's all rendered just the once & stored for all sessions - and now we don't need to call the whole function again when rendering in the content, too.

Reply:
🎉